### PR TITLE
aws.ec2 -Network Location sg_space referenced before assignment

### PR DESCRIPTION
**Pull request by ksteigerwald**
``` 
2019-02-28 03:12:32,627: custodian.aws:ERROR Error while executing policy
Traceback (most recent call last):
File "/src/c7n/policy.py", line 232, in run
resources = self.policy.resource_manager.resources()
File "/src/c7n/resources/ec2.py", line 93, in resources
return super(EC2, self).resources(query=query)
File "/src/c7n/query.py", line 419, in resources
resources = self.filter_resources(resources)
File "/src/c7n/manager.py", line 105, in filter_resources
resources = f.process(resources, event)
File "/src/c7n/filters/core.py", line 284, in process
return self.process_set(resources, event)
File "/src/c7n/filters/core.py", line 303, in process_set
resources = f.process(resources, event)
File "/src/c7n/filters/vpc.py", line 174, in process
found = self.process_resource(r, resource_sgs, resource_subnets, key)
File "/src/c7n/filters/vpc.py", line 232, in process_resource
sg_space != subnet_space):
UnboundLocalError: local variable 'sg_space' referenced before assignment
```

Resolves an issue where terminated instances lose their subnet and or security group details causing 
sg_space and subnet_space to be missing. Moving the variable decelerations up to ensure their set within function scope. 

Subclasses network location filter on EC2 to handle valid origin states. 

@thisisshi Big shout - DKHS

### CODE REVIEW

File: tests/test_ec2.py [hunk 1]
```diff
@@ -31,6 +31,33 @@
...
+            'terminated'
+        )
+
+        policy = self.load_policy(
+            {
+                'name': 'ec2-network-location',
+                'resource': 'ec2',
+                'filters': [
+                    {'State.Name': 'terminated'},
+                    {'type': 'network-location'}
```
Comment (kapilt) [inline tests/test_ec2.py]: this filter is missing a required attribute `key`, it shouldn't even be passing validation as written.

**Comment by kapilt** [comment]
its a bit unclear that the test here is actually verifying anything, fixing the test policy to be valid, and correcting for unrecorded files, its passes even with the implementation commented out.

**Comment by thisisshi** [comment]
Instantiating sg_space and subnet_space will fix the original traceback, subclassing the filter for ec2 was because the subnet and security group attributes (among others) get removed when the instance is terminated, returning incorrect results— since the subnets and sgs aren’t actually available on the resource, terminated shouldn’t be a valid state when filtering for network location (this was the reason why the error occurred in the first place).

**Comment by kapilt** [comment]
@thisisshi thanks, yeah that makes sense and sounds good. I realized I'd commented out ec2, not the vpc attrs when I was checking this.

**Comment by ksteigerwald** [comment]
@kapilt Thanks, I will add it shortly.

**Pull request merged**